### PR TITLE
feat(prek): supporting prek as an alternative to pre-commit and switching to prek

### PR DESCRIPTION
**Pull request by YazdanRa**
## Description

Replacing pre-commit with prek and add support for prek in hooks

## Checklist

- [X] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/)

### Code Changes

- [X] Add test cases to all the changes you introduce
- [X] Run `uv run poe all` locally to ensure this change passes linter check and tests
- [X] Manually test the changes:
  - [X] Verify the feature/bug fix works as expected in real-world scenarios
  - [X] Test edge cases and error conditions
  - [X] Ensure backward compatibility is maintained
  - [X] Document any manual testing steps performed
- [X] Update the documentation for the changes

### Documentation Changes

- [X] Run `uv run poe doc` locally to ensure the documentation pages renders correctly
- [X] Check and fix any broken links (internal or external) in the documentation

> When running `uv run poe doc`, any broken internal documentation links will be reported in the console output like this:
>
> ```text
> INFO    -  Doc file 'config.md' contains a link 'commands/bump.md#-post_bump_hooks', but the doc 'commands/bump.md' does not contain an anchor '#-post_bump_hooks'.
> ```

## Expected Behavior

Faster pre commit checks and hook installation 

## Steps to Test This Pull Request

N/A

## Additional Context

#1797

### CODE REVIEW

**Comment by YazdanRa** [comment]
I should maybe add the `pre-commit` in the `test` dependency group and parameterize the tests to check for both. wdyt?

**Comment by bearomorphism** [comment]
> I should maybe add the `pre-commit` in the `test` dependency group and parameterize the tests to check for both. wdyt?

sure, why not

File: commitizen/project_info.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 
 def is_pre_commit_installed() -> bool:
-    return bool(shutil.which("pre-commit"))
+    return bool(shutil.which("pre-commit")) or bool(shutil.which("prek"))
```
Comment (bearomorphism) [inline commitizen/project_info.py]: nit

```suggestion
    return any(shutil.which(tool) for tool in ("pre-commit", "perk"))
```

File: docs/tutorials/auto_check.md [hunk 2]
```diff
@@ -10,16 +10,22 @@ When a commit message fails validation, Git will reject the commit and display a
 
 There are two common methods for installing the hooks:
 
-### Method 1: Using [pre-commit](https://pre-commit.com/) (Recommended)
+### Method 1: Using [pre-commit](https://pre-commit.com/) or [prek](https://prek.j178.dev) (Recommended)
```
Comment (bearomorphism) [inline docs/tutorials/auto_check.md]:
```suggestion
### Method 1: Using pre-commit hook frameworks (Recommended)
```

File: docs/tutorials/auto_check.md [hunk 3]
```diff
@@ -10,16 +10,22 @@ When a commit message fails validation, Git will reject the commit and display a
...
 
-[pre-commit](https://pre-commit.com/) is a framework for managing and maintaining multi-language pre-commit hooks. It's the recommended approach as it handles hook installation, updates, and execution automatically.
+[pre-commit](https://pre-commit.com/) and [prek](https://prek.j178.dev) are frameworks for managing and maintaining multi-language pre-commit hooks. It's the recommended approach as it handles hook installation, updates, and execution automatically.
```
Comment (bearomorphism) [inline docs/tutorials/auto_check.md]:
```suggestion
Using pre-commit frameworks is the recommended approach as hook installation, updates, and execution are handled automatically.

These are two common pre-commit hooks

- [prek](https://prek.j178.dev) (faster)
- [pre-commit](https://pre-commit.com/)
```

File: docs/tutorials/auto_check.md [hunk 4]
```diff
@@ -10,16 +10,22 @@ When a commit message fails validation, Git will reject the commit and display a
...
 
 ```sh
 python -m pip install pre-commit
 ```
 
+or
+
+```sh
+python -m pip install prek
+```
```
Comment (bearomorphism) [inline docs/tutorials/auto_check.md:34]: I am not sure whether we should cover how to install those tools. User can check the official documentations for a comprehensive guide.

File: docs/tutorials/auto_check.md (hunk 2 above)
Comment (bearomorphism) [reply docs/tutorials/auto_check.md]: Or "pre-commit runner" or whatever you call those things

**Comment by bearomorphism** [comment]
Thanks!

File: docs/tutorials/auto_check.md (hunk 4 above)
Comment (YazdanRa) [reply docs/tutorials/auto_check.md:34]: I only kept the prek since it's faster.

File: docs/tutorials/auto_check.md [hunk 5]
```diff
@@ -10,16 +10,20 @@ When a commit message fails validation, Git will reject the commit and display a
 
 There are two common methods for installing the hooks:
 
-### Method 1: Using [pre-commit](https://pre-commit.com/) (Recommended)
+### Method 1: Using pre-commit hook frameworks (Recommended)
 
-[pre-commit](https://pre-commit.com/) is a framework for managing and maintaining multi-language pre-commit hooks. It's the recommended approach as it handles hook installation, updates, and execution automatically.
+Using pre-commit frameworks is the recommended approach as hook installation, updates, and execution are handled automatically.
+These are two common pre-commit hooks:
```
Comment (bearomorphism) [inline docs/tutorials/auto_check.md]:
```suggestion
These are two common pre-commit frameworks:
```

File: docs/tutorials/auto_check.md [hunk 6]
```diff
@@ -10,16 +10,20 @@ When a commit message fails validation, Git will reject the commit and display a
...
+- [prek](https://prek.j178.dev) (faster)
+- [pre-commit](https://pre-commit.com/)
```
Comment (bearomorphism) [inline docs/tutorials/auto_check.md:21]: Add something like "we will use perk throughout this section", and use AI to polish it if possible.

Thanks!

File: docs/tutorials/auto_check.md [hunk 7]
```diff
@@ -42,9 +46,10 @@ repos:
...
 
 ```bash
-pre-commit install --hook-type commit-msg
+prek install --hook-type commit-msg
 ```
 
+
 The hook is now active! Every time you create a commit, commitizen will automatically validate your commit message.
 
 ### Method 2: Manual Git hook installation
```
Comment (bearomorphism) [inline docs/tutorials/auto_check.md:63]: Could you help to check whether pre-commit (the framework) is still mentioned in the entire documentation site? If there is any, could you help to update if there's anything outdated.

Thanks!

File: docs/tutorials/auto_check.md (hunk 4 above)
Comment (bearomorphism) [reply docs/tutorials/auto_check.md:34]: sounds good

File: docs/tutorials/auto_check.md (hunk 7 above)
Comment (YazdanRa) [reply docs/tutorials/auto_check.md:63]: I couldn't find any other reference to the framework, there are still some mentions of pre-commit as the hooks which I left as is.

**Comment by bearomorphism** [comment]
This looks great! Let's wait for other maintainer's input.

File: docs/tutorials/auto_check.md [hunk 8]
```diff
@@ -2,22 +2,35 @@
...
-#### Step 1: Install pre-commit
+1. [prek](https://prek.j178.dev) (faster)
+2. [pre-commit](https://pre-commit.com/)
+
+
+In the steps below, we'll use `prek`.
+
+
+!!! tip "Using pre-commit framework"
+    Replace `prek` with `pre-commit` in the steps below if you prefer that tool. The configuration format is similar.
```
Comment (Lee-W) [inline docs/tutorials/auto_check.md]: I'm not sure I understand why this appears here 🤔  Might need some reworkding

Comment (YazdanRa) [reply docs/tutorials/auto_check.md]: I was thinking since the commands are the same, this tip can help users who rather using pre-commit to save some time instead of searching for pre-commit commands.

We can either remove it or maybe change it to something like:

> [!TIP]
> If you use **pre-commit** instead of **prek**, you can run the same commands.
> Simply replace `prek` with `pre-commit` in the steps below.

or 

> [!TIP]
> If you're already using **pre-commit**, the commands are identical.
> Replace `prek` with `pre-commit` and follow the same steps.

wdyt?

Comment (Lee-W) [reply docs/tutorials/auto_check.md]: I like the first one :)
